srcs/app/CGI/CgiHandler.cpp

```diff
@@ -0,0 +1,46 @@
+#include <CgiHandler.hpp>
+
+CgiHandler::CgiHandler(int socket, int cgi_output)
+	: socket_(socket), cgi_output_(cgi_output), cgi_complete_(false) {}
+
+CgiHandler::~CgiHandler() {
+	close(socket_);
+	close(cgi_output_);
+}
+
+int	CgiHandler::ReadCgiOutput() {
+	char	buffer[4096];
+
+	int nbytes = read(cgi_output_, buffer, sizeof(buffer));
```

Comment:
read (and write), return ssize_t, that is a long. Storing the return value into an int is a implicit downcast, and it's not a very advisable practice.

Reply:
Well spotted. I've changed it.

Reply:
There are some more instances of such...

Reply:
Thanks for pointing it out. I've changed it in `Connection.cpp` too.

I think now all the places that were using an int instead of ssize_t should be fixed.